templates/app/src/main/kotlin/group_id_folder/config/DynamicRemotePropertiesConfig.kt

```diff
@@ -7,4 +7,6 @@ import org.springframework.context.annotation.Configuration
 @RefreshScope
 @Configuration
 @ConfigurationProperties
-class DynamicRemotePropertiesConfig
+class DynamicRemotePropertiesConfig {
+    var prop: String? = null
+}
```

Comment:
```suggestion
@ConstructorBinding
data class DynamicRemotePropertiesConfig(val prop: String? = null)
```
Why not a data class?

Reply:
Adjusted as data class, but the annotation was @ConstructorBinding with the parameter "val prop" it was causing an error to initialize the application:

******************************
APPLICATION FAILED TO START
******************************

Description:

Failed to bind properties under '' to com.stackspot.balance.config.DynamicRemotePropertiesConfig$$EnhancerBySpringCGLIB$$2f14ecb9:

     Property: prop
     Value: prop-value1
     Origin: "prop" from property source "config/balance/"
     Reason: java.lang.IllegalStateException: No setter found for property: prop

Action:

Update your application's configuration


By setting the parameter to "val prop" the annotation became unnecessary, what do you think of this approach?